src/nupic/encoders/bigInt.cpp

```diff
@@ -0,0 +1,6509 @@
+// ==============================================================
+//
+//  Copyright (C) 1995  William A. Rossi
+//                      class RossiBigInt
+// 
+//  Copyright (C) 1999-2016  Alex Vinokur
+//                           class BigInt 
+//                           class BigInt::BaseBigInt 
+//                           class BigInt::Vin
+//                           upgrading class BigInt::Rossi
+//                           class BigInt::Run
+//                           class BigInt::Test
+//                           class BigInt::TestVin
+//                           class BigInt::TestRossi
+//                           class BigInt::PerformanceTestVin
+//                           class BigInt::PerformanceTestRossi 
+//
+//  ------------------------------------------------------------
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+//  ------------------------------------------------------------
+// 
+//  mailto:alex DOT vinokur AT gmail DOT com
+//  http://sourceforge.net/users/alexvn/
+//
+// ==============================================================
+
+
+// ##############################################################
+//
+//  SOFTWARE : Class BigInt
+//  FILE     : bigInt.cpp
+//
+//  DESCRIPTION : Implementation of the classes
```

Comment:
This is an interesting package.  It is rather large. is this something we can download as a dependancy?

Reply:
I aim to remove bigInt and md5, if feasible.

Reply:
note: it'll "break" the tests (as the hash changes), so we must make it all compile and pass, and then we can change the hash and claim the new values are changed just because of that, so accepted.